woocommerce/woocommerce-ios
# Order Creation: Refactor fees input

Part of: https://github.com/woocommerce/woocommerce-ios/issues/6027

## Description

This PR updates "Fee Details" screen to use better textfields handling (introduced in https://github.com/woocommerce/woocommerce-ios/pull/6268) and separate percentage/flat amount inputs.

## Changes

- Updates `FeeLineDetails` to use 2 different instances of `BindableTextfield`.
- Updates `FeeLineDetailsViewModel` to store percentage/fixed fees in independent variables.

## Testing

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. Tap "Create order" in the bottom sheet.
4. Tap "Add Product" button and select any product to enable "Payment" section.
5. Tap "Add Fees" button.
6. Check the Fee Details screen, enter a fixed amount and save.
7. Confirm that the "Payment" section displays correct fees and total values.
8. Tap the "Fees" row to open Fee Details, check that it's prefilled correctly.
9. Switch to the percentage fee type, check that it's prefilled with calculated value matching fixed amount.
10. Update percentage fee value and save.
11. Confirm that the "Payment" section displays correct fees and total values.
12. Tap the "Fees" row -> "Delete Fee from Order".
13. Confirm that the fees row is replaced with a button and the "Order Total" is correctly updated.

## Video

https://user-images.githubusercontent.com/3132438/155529570-90053b90-acc7-4660-b1f7-5c1ddaeeeb32.mp4

---
